tools/downloader/converter.py

```diff
@@ -40,35 +47,77 @@ def quote_windows(arg):
 else:
     quote_arg = shlex.quote
 
-def prefixed_printf(prefix, format, *args, **kwargs):
-    if prefix is None:
-        print(format.format(*args), **kwargs)
-    else:
-        print(prefix + ': ' + format.format(*args), **kwargs)
+def convert_to_onnx(context, model, output_dir, args):
+    context.printf('========= {}Converting {} to ONNX',
+                   '(DRY RUN) ' if args.dry_run else '', model.name)
+
+    conversion_to_onnx_args = [string.Template(arg).substitute(conv_dir=output_dir / model.subdirectory,
+                                                               dl_dir=args.download_dir / model.subdirectory)
+                               for arg in model.conversion_to_onnx_args]
+    cmd = [str(args.python), str(Path(__file__).absolute().parent / model.converter_to_onnx), *conversion_to_onnx_args]
+
+    context.printf('Conversion to ONNX command: {}', ' '.join(map(quote_arg, cmd)))
+
+    return True if args.dry_run else context.subprocess(cmd)
+
+
+class SingleThreadedContext(JobContext):
+    def printf(self, format, *args, flush=False):
+        print(format.format(*args), flush=flush)
 
-def prefixed_subprocess(prefix, args):
-    if prefix is None:
+    def subprocess(self, args):
         return subprocess.run(args).returncode == 0
 
-    with subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT,
-            universal_newlines=True) as p:
-        for line in p.stdout:
-            sys.stdout.write(prefix + ': ' + line)
-        return p.wait() == 0
 
+class BufferingRunner(JobContext):
```

Comment:
May be `MultiThreadedContext` more suitable? Or another name, mentioning about multi-threading?

Reply:
I ended up renaming both concrete context classes to more descriptive names.